Source/WebCore/html/HTMLInputElement.cpp

```diff
@@ -1043,8 +1044,11 @@ void HTMLInputElement::setChecked(bool isChecked)
 
     if (auto* buttons = radioButtonGroups())
         buttons->updateCheckedState(*this);
-    if (auto* renderer = this->renderer(); renderer && renderer->style().hasEffectiveAppearance())
+    if (auto* renderer = this->renderer(); renderer && renderer->style().hasEffectiveAppearance()) {
+        if (isSwitch())
+            downcast<CheckboxInputType>(*m_inputType).startSwitchAnimationTimer();
```

Comment:
I would call the method something like `performSwitchStateChangeAnimation` (or something along those lines) rather than `startSwitchAnimationTimer`. The timer is really an implementation detail.